python-checks/src/main/java/org/sonar/python/checks/ItemOperationsTypeCheck.java

```diff
@@ -44,23 +52,14 @@ public boolean isValidSubscription(Expression subscriptionObject, String require
     if (subscriptionObject.is(Tree.Kind.GENERATOR_EXPR)) {
       return false;
     }
-    if (subscriptionObject.is(Tree.Kind.CALL_EXPR)) {
-      Symbol subscriptionCalleeSymbol = ((CallExpression) subscriptionObject).calleeSymbol();
-      if (subscriptionCalleeSymbol != null && subscriptionCalleeSymbol.is(FUNCTION) && ((FunctionSymbol) subscriptionCalleeSymbol).isAsynchronous()) {
-        FunctionSymbol functionSymbol = (FunctionSymbol) subscriptionCalleeSymbol;
-        secondaries.put(functionSymbol.definitionLocation(), String.format(SECONDARY_MESSAGE, functionSymbol.name()));
-        return false;
-      }
+    if (subscriptionObject.is(Tree.Kind.CALL_EXPR) && isValidSubscriptionCallExpr((CallExpression) subscriptionObject, secondaries)) {
```

Comment:
seems that methods `isValidSubscriptionCallExpr` and `isValidSubscriptionSymbol` both have inverted logic to the `isValidSubscription`, so I'd suggest to rename them to `isInvalid...`